core/chains/evm/logpoller/log_poller.go

```diff
@@ -76,65 +76,50 @@ func NewLogPoller(orm *ORM, ec client.Client, lggr logger.Logger, pollPeriod tim
 		finalityDepth:     finalityDepth,
 		backfillBatchSize: backfillBatchSize,
 		addresses:         make(map[common.Address]struct{}),
-		topics:            make(map[int]map[common.Hash]struct{}),
+		eventSigs:         make(map[common.Hash]struct{}),
 	}
 }
 
-// MergeFilter will update the filter with the new topics and addresses.
+// MergeFilter will update the filter with the new eventSigs and addresses.
+// We do not support indexed topic filtering.
 // Clients may choose to MergeFilter and then replay in order to ensure desired logs are present.
-func (lp *logPoller) MergeFilter(topics []common.Hash, address common.Address) {
+func (lp *logPoller) MergeFilter(eventSigs []common.Hash, addresses []common.Address) {
 	lp.filterMu.Lock()
 	defer lp.filterMu.Unlock()
-	lp.addresses[address] = struct{}{}
-	// [[A, B], [C]] + [[D], [], [E]] = [[A, B, D], [C], [E]]
-	for i := 0; i < len(topics); i++ {
-		if lp.topics[i] == nil {
-			lp.topics[i] = make(map[common.Hash]struct{})
-		}
-		lp.topics[i][topics[i]] = struct{}{}
+	for _, address := range addresses {
+		lp.addresses[address] = struct{}{}
+	}
+	for _, eventSig := range eventSigs {
+		lp.eventSigs[eventSig] = struct{}{}
 	}
 }
 
-// Assumes caller holds filterMu lock
-func (lp *logPoller) filterAddresses() []common.Address {
-	var addresses []common.Address
+func (lp *logPoller) filter(from, to *big.Int, bh *common.Hash) ethereum.FilterQuery {
+	lp.filterMu.Lock()
+	defer lp.filterMu.Unlock()
+	var (
+		addresses []common.Address
+		eventSigs []common.Hash
+	)
 	for addr := range lp.addresses {
 		addresses = append(addresses, addr)
 	}
 	sort.Slice(addresses, func(i, j int) bool {
 		return bytes.Compare(addresses[i][:], addresses[j][:]) < 0
 	})
```

Comment:
Just for my understanding, but why is this sort necessary?

Reply:
its not required, imo just makes debug log reading easier e.g. comparing two logs which have overlapping filters